getting-started/configuration.md
# Configuration

## Config Files

This project uses `/src/lib/config/` as the config file directory, but most configurations do not need to be modified in general.

### site.ts | Site Info

```ts
export const site: SiteConfig = {
  protocol: import.meta.env.URARA_SITE_PROTOCOL ?? import.meta.env.DEV ? 'http://' : 'https://', // site protocol (usually no change is required)
  domain: import.meta.env.URARA_SITE_DOMAIN ?? 'urara-demo.netlify.app', // site domain
  title: 'Urara', // title
  subtitle: 'Sweet & Powerful SvelteKit Blog Template', // subtitle
  lang: 'en', // language
  descr: 'Powered by SvelteKit/Urara', // description
  author: {
    name: 'John Doe', // author name
    avatar: '/assets/maskable@512.png', // author image
    status: '🌸', // author status
    bio: 'lorem ipsum dolor sit amet, consectetur adipiscing elit.' // author bio
  },
  themeColor: '#3D4451' // theme color (currently only used in Manifest)
}
```

### icon.ts | Icons

Some icons are provided by default to be compatible with web app manifests and modern browsers, and can be replaced by yourself.

```
/urara/favicon.png - Website icon, 32x32
/urara/assets/any@180.png - Website icon, 180x180
/urara/assets/any@192.png - Website/Manifest icon, 192x192
/urara/assets/any@512.png - Manifest icon, 512x512
/urara/assets/manifest@192.png - Manifest mask icon, 192x192
/urara/assets/manifest@512.png - Manifest mask icon, 512x512
```

You can also replace the number of icons and their paths by modifying `/src/lib/config/icon.ts`.

### general.ts | Theme

```ts
export const theme: ThemeConfig = [
  {
    name: 'light', // Theme varable name
    text: '🌕 Light' // Theme display name, can be modified
  },
  {
    name: 'dark',
    text: '🌑 Dark'
  },
]
```

This project uses the [**daisyUI**](https://daisyui.com/) theme, see the availble [**daisyUI themes**](https://daisyui.com/docs/themes/?lang=en).

### general.ts | Header

```ts
export const header: HeaderConfig = {
  nav: [
    {
      text: 'Get Started', // Button name
      link: '/hello-world' // Link to page, or other sites
    },
    {
      text: 'Elements',
      link: '/hello-world/elements'
    }
  ]
}
```

Dropdown menu can added to the header following the format below.

```ts
{
   text: 'Hello World',
   children: [
     {
      text: 'Get Started',
      link: '/hello-world'
     },
     {
       text: 'Elements',
       link: '/hello-world/elements'
     },
	 {
	   text: 'ToC Disabled',
	   link: '/hello-world/toc-disabled'
	 }
   ]
}
```

### general.ts | Footer

Same as the Header, except the dropdown menu.

```ts
export const footer: FooterConfig = {
  nav: [
    {
      text: 'Feed', // Hypelink name
      link: '/atom.xml' // Link to page, or other sites
    },
    {
      text: 'Sitemap',
      link: '/sitemap.xml'
    }
  ]
}
```

### general.ts | Date Format

```ts
export const date: DateConfig = {
  locales: 'en', // Language, refer to IETF language tag
  options: {
    year: '2-digit', // Year: numeric / 2-digit
    weekday: 'long', // Week: narrow / short / long
    month: 'short', // Month: numeric / 2-digit / narrow / short / long
    day: 'numeric' // day: numeric / 2-digit
  }
}
```

For more additonal, see [**Intl.DateTimeFormat - JavaScript | MDN**](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat).


zh/getting-started/configuration.md
# 配置

## 配置文件

本項目使用 `/src/lib/config/` 作爲配置文件目錄，但多數配置在一般情況下無需修改。

### site.ts | 網站信息

```ts
export const site: SiteConfig = {
  protocol: import.meta.env.URARA_SITE_PROTOCOL ?? import.meta.env.DEV ? 'http://' : 'https://', // 協議（一般無需更改）
  domain: import.meta.env.URARA_SITE_DOMAIN ?? 'urara-demo.netlify.app', // 域名
  title: 'Urara', // 標題
  subtitle: 'Sweet & Powerful SvelteKit Blog Template', // 副標題
  lang: 'zh-TW', // 語言
  descr: 'Powered by SvelteKit/Urara', // 描述
  author: {
    name: 'John Doe', // 作者名稱
    avatar: '/assets/maskable@512.png', // 作者圖片
    status: '🌸', // 作者狀態
    bio: 'lorem ipsum dolor sit amet, consectetur adipiscing elit.' // 作者描述
  },
  themeColor: '#3D4451' // 主題顔色（目前僅用於 Manifest）
}
```

### icon.ts | 圖示

默認提供一些圖示以兼容 Web app manifests 及現代瀏覽器，可以自行替換。

```
/urara/favicon.png - 網站圖示，32x32
/urara/assets/any@180.png - 網站圖示，180x180
/urara/assets/any@192.png - 網站 / Manifest 圖示，192x192
/urara/assets/any@512.png - Manifest 圖示，512x512
/urara/assets/manifest@192.png - Manifest 遮罩圖示，192x192
/urara/assets/manifest@512.png - Manifest 遮罩圖示，512x512
```

也可以通過修改 `/src/lib/config/icon.ts` 替換圖示數量及路徑。

### general.ts | 主題

```ts
export const theme: ThemeConfig = [
  {
    name: 'light', // 主題變量名
    text: '🌕 Light' // 主題顯示名，可修改
  },
  {
    name: 'dark',
    text: '🌑 Dark'
  },
]
```

本項目使用了 [**daisyUI**](https://daisyui.com/) 主題，查看可用的 [**daisyUI 主題**](https://daisyui.com/docs/themes/?lang=zh_tw)。

### general.ts | 標題欄

```ts
export const header: HeaderConfig = {
  nav: [
    {
      text: 'Get Started', // 按鈕名稱
      link: '/hello-world' // 指向的頁面，可以填寫其他網站
    },
    {
      text: 'Elements',
      link: '/hello-world/elements'
    }
  ]
}
```
還可以根據下面的格式在標題欄添加下拉選單。

```ts
{
   text: 'Hello World',
   children: [
     {
      text: 'Get Started',
      link: '/hello-world'
     },
     {
       text: 'Elements',
       link: '/hello-world/elements'
     },
	 {
	   text: 'ToC Disabled',
	   link: '/hello-world/toc-disabled'
	 }
   ]
}
```

### general.ts | 頁腳

大致與標題相同，下拉選單除外。

```ts
export const footer: FooterConfig = {
  nav: [
    {
      text: 'Feed', // 超連結名稱
      link: '/atom.xml' // 指向的頁面，可以填寫其他網站
    },
    {
      text: 'Sitemap',
      link: '/sitemap.xml'
    }
  ]
}
```

### general.ts | 日期格式

```ts
export const date: DateConfig = {
  locales: 'zh-TW', // 語言，可參考 IETF 語言標籤
  options: {
    year: '2-digit', // 年份: numeric / 2-digit
    weekday: 'long', // 星期: narrow / short / long
    month: 'short', // 月份: numeric / 2-digit / narrow / short / long
    day: 'numeric' // 日期: numeric / 2-digit
  }
}
```

要添加更多選項，請參考 [**Intl.DateTimeFormat - JavaScript | MDN**](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat)。


.vitepress/config.ts
import { defineConfig } from 'vitepress'
import { withPwa } from '@vite-pwa/vitepress'

const nav = (_lang, index) => [
  {
    text: ['Discussions', '討論', '', '讨论'][index],
    link: 'https://github.com/importantimport/urara/discussions',
  },
]

const sidebar = (lang, index) => [
  {
    text: ['Getting Started', '入門', 'はじめに', '入门'][index],
    collapsible: true,
    items: [
      {
        text: ['Intro', '介紹', '紹介', '介绍'][index],
        link: lang + 'getting-started/intro',
      },
      {
        text: ['Quick Start', '快速開始', 'クイックスタート', '快速开始'][
          index
        ],
        link: lang + 'getting-started/quick-start',
      },
      {
        text: ['Configuration', '配置', '設定', '配置'][index],
        link: lang + 'getting-started/configuration',
      },
      {
        text: ['Writing', '寫作', '文章を書く', '写作'][index],
        link: lang + 'getting-started/writing',
      },
      {
        text: ['Deploy', '部署', '配備', '部署'][index],
        link: lang + 'getting-started/deploy',
      },
    ],
  },
  {
    text: ['Advanced', '進階', '高度な', '进阶'][index],
    collapsible: true,
    items: [
      {
        text: ['Extension', '拓展', '拡張', '拓展'][index],
        link: lang + 'advanced/extension',
      },
    ],
  },
]

const themeConfig = (lang, index) => ({
  sidebar: sidebar(lang, index),
  nav: nav(lang, index),
  editLink: {
    pattern: 'https://github.com/importantimport/urara-docs/edit/main/:path',
    text: [
      'Edit on GitHub',
      '在 GitHub 上編輯',
      'GitHub でこのページを編集',
      '在 GitHub 上编辑',
    ][index],
  },
  lastUpdatedText: ['Last Updated', '最後更新于', '最終更新', '最后更新于'][
    index
  ],
})

export default withPwa(
  defineConfig({
    title: 'Urara Docs',
    head: [
      [
        'link',
        {
          rel: 'icon',
          href: '/favicon.png',
          sizes: '32x32',
          type: 'image/png',
        },
      ],
      [
        'script',
        {
          src: 'https://plausible.kwaa.dev/js/plausible.js',
          defer: '',
          'data-domain': 'urara-docs.netlify.app',
        },
      ],
    ],
    description: 'Sweet & Powerful SvelteKit Blog Template.',
    markdown: { lineNumbers: true },
    locales: {
      root: {
        lang: 'en-US',
        label: 'English',
        themeConfig: themeConfig('/', 0),
      },
      zh: {
        lang: 'zh-TW',
        label: '正體中文',
        themeConfig: themeConfig('/zh/', 1),
      },
      // ja: { lang: 'ja-JP', label: '日本語', themeConfig: themeConfig('/ja/', 2) }
      'zh-hans': {
        lang: 'zh-Hans',
        label: '简体中文',
        themeConfig: themeConfig('/zh-hans/', 3),
      },
    },
    themeConfig: {
      socialLinks: [
        { icon: 'github', link: 'https://github.com/importantimport/urara' },
      ],
    },
    pwa: {
      manifest: {
        name: 'Urara',
        short_name: 'Urara',
        description: 'Sweet & Powerful SvelteKit Blog Template.',
        theme_color: '#ffffff',
        icons: [
          {
            src: 'assets/any@192.png',
            sizes: '192x192',
            type: 'image/png',
          },
          {
            src: 'assets/any@512.png',
            sizes: '512x512',
            type: 'image/png',
          },
          {
            src: 'assets/maskable@192.png',
            sizes: '192x192',
            type: 'image/png',
            purpose: 'any maskable',
          },
          {
            src: 'assets/maskable@512.png',
            sizes: '512x512',
            type: 'image/png',
            purpose: 'any maskable',
          },
        ],
      },
    },
  })
)
